frontend/src/metabase/admin/tasks/components/LogLevelsModal/LogLevelsModal.tsx

```diff
@@ -0,0 +1,165 @@
+import { type FormEvent, useEffect, useState } from "react";
+import { t } from "ttag";
+
+import {
+  useAdjustLogLevelsMutation,
+  useListLoggerPresetsQuery,
+  useResetLogLevelsMutation,
+} from "metabase/api/logger";
+import { CodeEditor } from "metabase/components/CodeEditor";
+import { LoadingAndErrorWrapper } from "metabase/components/LoadingAndErrorWrapper";
+import ModalContent from "metabase/components/ModalContent";
+import { Box, Button, Flex, Icon, Loader, Tooltip } from "metabase/ui";
+import type { TimeUnit } from "metabase-types/api";
+
+import { DurationInput } from "./DurationInput";
+import S from "./LogLevelsModal.module.css";
+import { PresetPicker } from "./PresetPicker";
+import {
+  getErrorMessage,
+  getLogLevelsErrorMessage,
+  getPresetJson,
+  isJsonValid,
+} from "./utils";
+
+interface Props {
+  onClose: () => void;
+}
+
+export const LogLevelsModal = ({ onClose }: Props) => {
+  const {
+    data: presets = [],
+    error: presetsError,
+    isLoading: isLoadingPresets,
+  } = useListLoggerPresetsQuery();
+  const [adjust, { error: adjustError, isLoading: isLoadingAdjust }] =
+    useAdjustLogLevelsMutation();
+  const [reset, { error: resetError, isLoading: isLoadingReset }] =
+    useResetLogLevelsMutation();
+
+  const [duration, setDuration] = useState("60");
+  const [durationUnit, setDurationUnit] = useState<TimeUnit>("minutes");
+  const [json, setJson] = useState("");
+  const durationNumber = parseInt(duration, 10);
+  const isDurationValid = Number.isFinite(durationNumber);
+  const isValid = isDurationValid && isJsonValid(json);
+  const isLoading = isLoadingAdjust || isLoadingReset;
+
+  const handleReset = async () => {
+    const response = await reset();
+
+    if (!response.error) {
+      onClose();
+    }
+  };
+
+  const handleSubmit = async (event: FormEvent) => {
+    event.preventDefault();
+
+    const response = await adjust({
+      duration: durationNumber,
+      duration_unit: durationUnit,
+      log_levels: JSON.parse(json),
+    });
+
+    if (!response.error) {
+      onClose();
+    }
+  };
+
+  useEffect(
+    function autoApplyFirstPreset() {
+      if (presets.length > 0) {
+        setJson(getPresetJson(presets[0]));
+      }
+    },
+    [presets],
+  );
+
+  if (presetsError || isLoadingPresets) {
+    return (
+      <ModalContent title={t`Customize log levels`} onClose={onClose}>
+        <LoadingAndErrorWrapper
+          error={presetsError}
+          loading={isLoadingPresets}
+        />
+      </ModalContent>
+    );
+  }
+
+  return (
+    <ModalContent title={t`Customize log levels`} onClose={onClose}>
+      <form onSubmit={handleSubmit}>
```

Comment:
You might want to use our `metabase/forms` package for that. I'll get validation & error handling on submit for free.

Reply:
8e6c4a0828bf3afd46e2ca4ae508079b3761bdaa